tests/test_logging.py

```diff
@@ -26,25 +26,47 @@ def test_setOverrideLogLevel(self):
         self.assertEqual(notherlogger.getEffectiveLevel(), logging.ERROR)
         setOverrideLogLevel('INFO')
         somelogger = getLogger('foo.bar')
-        setOverrideLogLevel('INFO')
-
-
-class TestLogging2(TestCase):
-
-    def setUp(self):
-        initLogging()
 
     def test_getLevelName(self):
         self.assertEqual(getLevelName('ERROR'), logging.ERROR)
         self.assertEqual(getLevelName('FATAL'), logging.ERROR)
         self.assertEqual(getLevelName('OFF'), logging.CRITICAL)
+        setOverrideLogLevel('INFO')
+
+
+class TestLoggingConfiguration(TestCase):
 
-    def test_configfile(self):
+    def test_tmpConfigfile(self):
+        self.assertNotEqual(logging.getLogger('').getEffectiveLevel(), logging.NOTSET)
         with TemporaryDirectory() as tempdir:
             with pushd_popd(tempdir):
-                with open('ocrd_logging.py', 'w') as f:
-                    f.write('print("this is mighty dangerous")')
+                with open('ocrd_logging.conf', 'w') as f:
+                    # write logging configuration file (MWE)
+                    f.write('''
+                        [loggers]
+                        keys=root
+                        
+                        [handlers]
+                        keys=consoleHandler
+                        
+                        [formatters]
+                        keys=
+                        
+                        [logger_root]
+                        level=ERROR
+                        handlers=consoleHandler
+                        
+                        [handler_consoleHandler]
+                        class=StreamHandler
+                        formatter=
+                        args=(sys.stdout,)
```

Comment:
Is it really necessary to have all these instructions? How does my minimal proposal fail?

Reply:
The parser fails with `KeyErrors` like 
```
tests/test_logging.py:64: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ocrd_utils/ocrd_utils/logging.py:93: in initLogging
    logging.config.fileConfig(config_file)
/usr/lib/python3.6/logging/config.py:76: in fileConfig
    formatters = _create_formatters(cp)
/usr/lib/python3.6/logging/config.py:109: in _create_formatters
    flist = cp["formatters"]["keys"]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <configparser.ConfigParser object at 0x7f7157cb60f0>, key = 'formatters'

    def __getitem__(self, key):
        if key != self.default_section and not self.has_section(key):
>           raise KeyError(key)
E           KeyError: 'formatters'

/usr/lib/python3.6/configparser.py:959: KeyError
```

But it's fine adding a rather empty section. Also it complains where there's no handler set for rootLogger, which in turn means that there must be a handler configured, which has to have attribute `class`set and so forth.

Reply:
I see. So your version _is_ the minimum.